### PR TITLE
Introduce a test suite for the assembler example

### PR DESCRIPTION
**Pull request by grigory-rechistov**
Fix a few discovered issue along the way.

### CODE REVIEW

File: examples/xed-asmparse-main.c [hunk 1]
```diff
@@ -356,6 +355,29 @@ static void set_mode_vec(xed_reg_enum_t reg,
...
+{
+    if (*s == '+' || *s == '-') /* skip leading sign */
+        s++;
+    if (*s == '0' && *(s + 1) == 'X') /* skip hexadecimal prefix */
+        s += 2;
+    return (*s == '0');
+}
+
+/* A nibble is a 4 bits wide hexadecimal digit.
+   Note that decimal digits are not nibbles (approx 3.2 bits) but
```
Comment (markcharney) [inline examples/xed-asmparse-main.c]: no need for this 3.2 bits comment. pls  remove.

File: examples/xed-asmparse-main.c [hunk 2]
```diff
@@ -391,7 +413,16 @@ static void process_operand(xed_enc_line_parsed_t* v,
         }
     }
     else if (q->type == OPND_IMM) {
-        xed_uint_t nbits = get_nbits_signed(q->imm);
+		/* If user padded the number with leading zeroes, consider this to be
```
Comment (markcharney) [inline examples/xed-asmparse-main.c]: I think a TAB char snuck in here; no tab chars in source code.

File: examples/xed-asmparse-main.c [hunk 3]
```diff
@@ -582,7 +582,10 @@ static void encode_with_xed(xed_enc_line_parsed_t* v)
     q = v->opnds;
     while(q) {
         process_operand(v, q, &noperand, operand_array, &has_imm0, &eosz);
-        assert(noperand < ASP_MAX_OPERANDS);
+        if (noperand >= ASP_MAX_OPERANDS) {
+            asp_error_printf("Too many operands\n");
+            exit(1);
+        }
         q = q->next;
     }
```
Comment (markcharney) [inline examples/xed-asmparse-main.c]: separate code changes & tests.

File: examples/xed-asmparse-main.c [hunk 4]
```diff
@@ -30,7 +30,9 @@ END_LEGAL */
 
 #define ASP_MAX_OPERANDS (XED_ENCODER_OPERANDS_MAX+3)
 
-static void process_args(int argc, char** argv, xed_enc_line_parsed_t* v, int* verbose) {
+static void process_args(int argc, char** argv, xed_enc_line_parsed_t* v,
+                         int* verbose, int* quiet) {
```
Comment (markcharney) [inline examples/xed-asmparse-main.c]: not sure I am psyched about having "verbose" and "quiet" separate variables.  I think I am more comfortable thinking of:

"quiet" as verbose=0, 
"normal" as verbose=1 (default) and
"verbose" as verbose > 1
"debug" as verbose > 2

Would you mind trying that approach instead?

Comment (grigory-rechistov) [reply examples/xed-asmparse-main.c]: Will introduce verbosity levels then

**Pull request closed without merging**